workers/unity/Assets/Playground/Scripts/Cubes/ScoreCollision.cs

```diff
@@ -0,0 +1,52 @@
+using Improbable.Gdk.Core;
+using UnityEngine;
+using Unity.Entities;
+using Generated.Playground;
+using Unity.Mathematics;
+
+[RemoveAtEndOfTick]
+public struct CollisionComponent : IComponentData
+{
+    public SpatialOSLaunchable other;
+}
+
+public class ScoreCollision : MonoBehaviour
+{
+    private MutableView view;
+    private SpatialOSLaunchable ownLaunchable;
+
+    public void Start()
+    {
+        var component = GetComponent<SpatialOSComponent>();
+        if (component)
+        {
+            view = WorkerRegistry.GetWorkerForWorld(component.World).View;
+            ownLaunchable = view.GetComponent<SpatialOSLaunchable>(component.Entity);
+        }
+        else
+        {
+            Debug.LogError("Could not retrieve SpatialOSComponent from game object.");
+        }
+    }
+
+    public void OnCollisionEnter(Collision col)
+    {
+        if (col.gameObject && col.gameObject.tag == "Cube")
+        {
+            var component = col.gameObject.GetComponent<SpatialOSComponent>();
+            if (component != null && view.HasComponent(component.Entity, typeof(SpatialOSLaunchable)))
+            {
+                SpatialOSLaunchable otherEntity = view.GetComponent<SpatialOSLaunchable>(component.Entity);
+                var entityManager = World.Active.GetExistingManager<EntityManager>();
+                Entity collisionEventEntity = entityManager.CreateEntity(typeof(SpatialOSLaunchable), typeof(CollisionComponent),
```

Comment:
Why are you creating a new entity, could the component be added to an entity which already exists? Also if this is required when do the entities get cleaned up?

Reply:
Indeed it can, it's two different ways of approaching it.
The entities get cleaned up because of the [RemoveAtEndOfTick] annotation of CollisionComponent.

Reply:
The `RemoveAtEndOfTick` doesn't clean up the entity - just the component

Reply:
I mean yes this approach does work but its much more heavy weight, you are creating an entity with 3 components but you could just be adding a component to an existing entity.